components/brave_ads/core/internal/account/utility/redeem_unblinded_token/create_confirmation_url_request_builder.cc

```diff
@@ -54,15 +54,18 @@ mojom::UrlRequestInfoPtr CreateConfirmationUrlRequestBuilder::Build() {
 ///////////////////////////////////////////////////////////////////////////////
 
 GURL CreateConfirmationUrlRequestBuilder::BuildUrl() const {
+  const std::string host = confirmation_.ad_type == AdType::kSearchResultAd
+                               ? server::GetAnonymousSearchHost()
+                               : server::GetAnonymousHost();
+
   std::string credential_base64_url;
   if (confirmation_.opted_in && confirmation_.opted_in->credential_base64url) {
     credential_base64_url =
         base::StrCat({"/", *confirmation_.opted_in->credential_base64url});
   }
 
   const std::string spec = base::StringPrintf(
-      "%s/v3/confirmation/%s%s",
-      GetAnonymousHost(confirmation_.ad_type).c_str(),
+      "%s/v3/confirmation/%s%s", host.c_str(),
       confirmation_.transaction_id.c_str(), credential_base64_url.c_str());
 
   return GURL(spec);
```

Comment:
do we have a DCHECK that returning url is valid?

Reply:
Added a DCHECK to ads_service_impl